src/mscorlib/src/System/Span.cs

```diff
@@ -612,7 +612,11 @@ internal static unsafe void ClearWithoutReferences(ref byte b, nuint byteLength)
         {
             if (byteLength == 0)
                 return;
-
+            
+#if AMD64
+            Unsafe.InitBlockUnaligned(ref b, 0, (uint)byteLength);
```

Comment:
One way to fix it would be to call: InitBlockUnaligned for smaller sizes (say, several kB) and PInvoke for sizes over that.